warehouse/accounts/services.py

```diff
@@ -77,6 +79,20 @@ def get_user(self, userid):
         # TODO: We need some sort of Anonymous User.
         return self.db.query(User).get(userid)
 
+    def get_two_factor(self, user_id):
+        result = self.db.query(TwoFactor).filter(TwoFactor.user_id == user_id).first()
+
+        if result:
+            return result
+        else:
+            user = self.get_user(user_id)
+            if not user:
+                return None
+            two_factor = TwoFactor(user=user)
+            self.db.add(two_factor)
+            self.db.flush()
```

Comment:
Is this flush really required? It might be, but normally SQLAlchemy should handle flushing as needed behind the scenes.

Reply:
I added that `flush` based on `create_user` and `add_email`, although both of those perform the explicit flush to ensure that the `id` attribute is immediately available. I'll try removing it here and seeing if anything explodes 😄